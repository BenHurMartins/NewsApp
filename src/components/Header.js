import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import {Colors, Mixins} from '../styles';
import Icon from 'react-native-vector-icons/AntDesign';
Icon.loadFont();
const Header = ({title, leftAction}) => {
  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.sideContainer} onPress={leftAction}>
        {leftAction ? (
          <Icon
            name={'left'}
            color={Colors.WHITE}
            size={Mixins.FONT_SIZE_BIG}
          />
        ) : null}
      </TouchableOpacity>
      <Text style={styles.title}>{title}</Text>
      <View style={styles.sideContainer}></View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    height: 60,
    backgroundColor: Colors.PRIMARY,
    width: Mixins.DEVICE_WIDTH,
    alignItems: 'center',
    flexDirection: 'row',
  },
  title: {
    color: Colors.WHITE,
    fontSize: Mixins.FONT_SIZE_BIG,
    fontWeight: '500',
    width: Mixins.DEVICE_WIDTH80,
    textAlign: 'center',
  },
  sideContainer: {
    width: Mixins.DEVICE_WIDTH10,
    justifyContent: 'center',
    alignItems: 'center',
  },
});
