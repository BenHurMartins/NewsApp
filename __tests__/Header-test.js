import 'react-native';
import React from 'react';
import {render, act} from '@testing-library/react-native';
import Header from '../src/components/Header';
import renderer from 'react-test-renderer';

describe('Header component', () => {
  it('Header renders correctly', () => {
    renderer.create(<Header title={'Test'} />);
  });
  it('Header snapshot test', () => {
    const tree = renderer.create(<Header title={'Test'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Should set the title', () => {
    const {getByText} = render(<Header title={'Test'} />);
    act(() => {
      expect(getByText('Test')).not.toBe(null);
    });
  });
});
