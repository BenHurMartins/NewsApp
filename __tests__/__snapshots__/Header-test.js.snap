// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#65a1dd",
      "flexDirection": "row",
      "height": 60,
      "width": 750,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
        "width": 75,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 37.5,
        "fontWeight": "500",
        "textAlign": "center",
        "width": 600,
      }
    }
  >
    Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": 75,
      }
    }
  />
</View>
`;
